### PR TITLE
Add navbar button to copy image

### PR DESCRIPTION
**Pull request by bo0tzz**
Todo:

- [x] `photoViewer` field in asset-viewer.svelte is null after navigating to another asset
- [x] Add 'photo copied' popup notification

### CODE REVIEW

**Comment by bo0tzz** [comment]
@alextran1502 can you take a look at that null issue?

**Review by alextran1502 — COMMENTED** [review]
I tested this and it works fine. You can merge this in. Probably added a notification when we click the button so the user know what just happened

**Comment by alextran1502** [comment]
Ah hold on, I see the issue you are mentioned

**Comment by alextran1502** [comment]
@bo0tzz 
The problem was because we use the `{#key asset.id}`, which will destroy a component when the value of `asset.id` change and rerender it. Therefore it binds to null.

I fixed it by using a global event system. The catch is I am unsure how to make it strong type so that it doesn't show squiggly line T_T.

This is now ready to merge to main

**Comment by bo0tzz** [comment]
Awesome, thanks a lot! I'll take a final look later this week and add the notification popup.

File: web/src/app.d.ts [hunk 1]
```diff
@@ -9,3 +9,11 @@ declare namespace App {
...
 	// interface Platform {}
 }
+
+// Source: https://stackoverflow.com/questions/63814432/typescript-typing-of-non-standard-window-event-in-svelte
+// To fix the <svelte:window... in components/asset-viewer/photo-viewer.svelte
+declare namespace svelte.JSX {
+	interface HTMLAttributes<T> {
+		oncopyImage?: () => void;
+	}
+}
```
Comment (bo0tzz) [inline web/src/app.d.ts:19]: Is this the right way to fix that type issue? Feels like I'm doing some global thing here, but maybe that's just how it is.

**Review by bo0tzz — COMMENTED** [review]
One typescript question, other than that this one is ready to merge from my end :)

File: web/src/app.d.ts (hunk 1 above)
Comment (bo0tzz) [reply web/src/app.d.ts]: Turns out this fixes the typescript error but causes an eslint one:
```
/home/runner/work/immich/immich/web/src/app.d.ts
  16:12  warning  'HTMLAttributes' is defined but never used  @typescript-eslint/no-unused-vars
  16:27  warning  'T' is defined but never used               @typescript-eslint/no-unused-vars
```

**Review by alextran1502 — APPROVED** [review]
LGTM, do you want to add the popup when the action is successfully performed?

**Comment by bo0tzz** [comment]
That popup should be in there already! Unless I messed something up lol

**Comment by alextran1502** [comment]
> That popup should be in there already! Unless I messed something up lol

I'm blind today, merging

**Pull request merged**